src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,110 @@
+    private static final String DEFAULT_LOGIN = "user_login";
+    private static RegistrationService registrationService;
+    private static StorageDaoImpl storageDao;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUser() {
+        user = new User();
+        user.setLogin(DEFAULT_LOGIN);
+        user.setPassword(DEFAULT_PASSWORD);
+        user.setAge(DEFAULT_AGE);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void ageLessThan18_notOK() {
+        user.setAge(DEFAULT_AGE - 1);
+        assertThrows(RegistrationException.class,() -> registrationService.register(user));
+    }
+
+    @Test
+    void register_addNullUser_notOk() {
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(null),
+                "User can't be null");
+    }
+
+    @Test
+    void register_addNullLogin_notOk() {
+        user.setLogin(null);
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(user),
+                "User`s login can't be null");
+    }
+
+    @Test
+    void register_addNullAge_notOk() {
+        user.setAge(null);
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(user),
+                "User`s age can't be null");
+    }
+
+    @Test
+    void register_addNullPassword_notOk() {
+        user.setPassword(null);
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(user),
+                "User`s password can't be null");
+    }
+
+    @Test
+    void validUser_Ok() {
+        registrationService.register(user);
+        User expected = storageDao.get(user.getLogin());
+        Assertions.assertEquals(expected, user, "Age field's passed");
+    }
+
+    @Test
+    void register_passwordLengthIsLessThanMinLength_notOk() {
+        user.setPassword("321");
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(user),
+                "User`s password should be 6 or more symbols");
+    }
+
+    @Test
+    void register_passwordLengthIsLongerThanMinLength_ok() {
+        user.setPassword(DEFAULT_PASSWORD + "1");
+        registrationService.register(user);
+        User actual = storageDao.get(user.getLogin());
+        assertEquals(user, actual, "All user`s fields are correct");
+    }
+
+    @Test
+    void register_loginExists_notOk() {
+        registrationService.register(user);
+        registrationService.register(user);
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(user),
+                "User is already exists");
+    }
+}
```

Comment:
messages in assert are shown when test fails, so you need to show info that helps us understand why test fail and what is expected behavior
ex: "should throw OurException when user is null"
rewrite all messages and make them informative

also you need tests for all edge case scenarious: password length 5/6/7  age 17/18/19